### PR TITLE
chore(external docs): Add note about trace events to concepts page

### PR DESCRIPTION
**Pull request by neuronull**
User in discord support was asking about trace support in vector.

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]

File: website/content/en/docs/about/concepts.md [hunk 1]
```diff
@@ -24,6 +24,11 @@ A **metric** event represents a numerical operation performed on a time series.
 
 {{< jump "/docs/about/under-the-hood/architecture/data-model/metric" >}}
 
+### Traces
+
+A **trace** event can be thought of as a special kind of log event. Currently Vector only supports trace events for the `datadog_agent` source, and the `datadog_traces` sink.
```
Comment (spencergilbert) [inline website/content/en/docs/about/concepts.md:29]: technically there's some transforms that support, not sure what you wanted to list?

Comment (neuronull) [reply website/content/en/docs/about/concepts.md:29]: would that be Sample and Remap ?

Comment (spencergilbert) [reply website/content/en/docs/about/concepts.md:29]: IIRC, yes 😅

Comment (neuronull) [reply website/content/en/docs/about/concepts.md:29]: 🤔 My intent with that line was to better communicate _which_ components we already have support for that in 😅

Comment (spencergilbert) [reply website/content/en/docs/about/concepts.md:29]: Yeah, balance between knowing for sure that list is going to get out of date and being more useful for the users.

Ideally the site would be rendering what event types each component can handle, it used to but doesn't anymore.

**Review by spencergilbert — APPROVED** [review]

File: website/content/en/docs/about/concepts.md (hunk 1 above)
Comment (neuronull) [reply website/content/en/docs/about/concepts.md:29]: Perhaps a medium could be something like:

"At this writing, the components that support Trace events are <...>". If you're interested in using traces with Vector and have a component that doesn't yet support them, please open an issue so we can better prioritize adding support."

File: website/content/en/docs/about/concepts.md [hunk 2]
```diff
@@ -24,6 +24,11 @@ A **metric** event represents a numerical operation performed on a time series.
 
 {{< jump "/docs/about/under-the-hood/architecture/data-model/metric" >}}
 
+### Traces
+
+A **trace** event can be thought of as a special kind of log event. As of this writing, the components that support trace events are: the `datadog_agent` source, the `datadog_traces` sink, and the `sample` and `remap` transforms.
+If you're interested in using traces with a Vector component that doesn't yet support them, please open an issue so we can have a better understanding of what components to prioritize adding trace support for."
```
Comment (spencergilbert) [inline website/content/en/docs/about/concepts.md:30]:
```suggestion
A **trace** event can be thought of as a special kind of log event. As of this writing, the components that support trace events are: the `datadog_agent` source, the `datadog_traces` sink, and the `sample` and `remap` transforms.

If you're interested in using traces with a Vector component that doesn't yet support them, please open an issue so we can have a better understanding of what components to prioritize adding trace support for."
```

Line break is probably going to render more nicely 👍

**Review by spencergilbert — COMMENTED** [review]
We're not ready to add anything [here](https://deploy-preview-16315--vector-project.netlify.app/docs/about/under-the-hood/architecture/data-model/) right?

**Comment by neuronull** [comment]
> We're not ready to add anything [here](https://deploy-preview-16315--vector-project.netlify.app/docs/about/under-the-hood/architecture/data-model/) right?

I'd say no, simply because traces are currently more or less equivalent to log events as far as vector is concerned.
But I think if (when) we get to that point where we more concretely define them and their distinguishing characteristics, that would be a good time to extend that section.

But, that is just my 2c :shrug:

**Review by spencergilbert — APPROVED** [review]
Matches my understanding at this point 👍

**Comment by neuronull** [comment]
Does this need super powers for merging in ?

**Comment by neuronull** [comment]
❤️

**Pull request merged**